internal/plugins/golang/v2/config.go

```diff
@@ -14,16 +14,20 @@
 
 package v2
 
-import "sigs.k8s.io/kubebuilder/v2/pkg/model/config"
+import (
+	"sigs.k8s.io/kubebuilder/v3/pkg/config"
+	cfgv3 "sigs.k8s.io/kubebuilder/v3/pkg/config/v3"
+)
 
 // Config configures this plugin, and is saved in the project config file.
 type Config struct{}
 
-// hasPluginConfig returns true if cfg.Plugins contains an exact match for this plugin's key.
-func hasPluginConfig(cfg *config.Config) bool {
-	if !cfg.IsV3() || len(cfg.Plugins) == 0 {
+// hasPluginConfig returns true if cfg.Layout contains an exact match for this plugin's key.
+func hasPluginConfig(cfg config.Config) bool {
+	isV3 := cfg.GetVersion().Compare(cfgv3.Version) >= 0
+	if !isV3 {
 		return false
 	}
```

Comment:
`cfg.DecodePluginConfig(...)` for config/v2 or lower will always return an error, so it will always be different to `nil`, so it is already false, no need to add the check before.
```suggestion
```